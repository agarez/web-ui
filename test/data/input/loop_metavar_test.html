<!DOCTYPE html>
<!--
Copyright (c) 2013, the Dart project authors.  Please see the AUTHORS file
for details. All rights reserved. Use of this source code is governed by a
BSD-style license that can be found in the LICENSE file.
-->
<html lang="en">
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
  <script src="packages/web_ui/testing/testing.js"></script>
</head>
<body>
  Test the $index meta variable:
  <template repeat="item in items" indentation="remove">
    <div>
    "{{item}}" is the {{$index}}{{suffix($index)}} item out of {{$list.length}}.
    </div>
  </template>
  <script type="application/dart">
  import 'dart:html';
  import 'package:web_ui/observe.dart';

  @observable
  List items = toObservable(['foo', 'bar', 'baz', 'qux', 'quux']);

  String suffix(int index) {
    switch (index) {
      case 1: return 'st';
      case 2: return 'nd';
      case 3: return 'rd';
      default: return 'th';
    }
  }

  main() {
    window.postMessage('done', '*');
  }
  </script>
</body>
</html>
