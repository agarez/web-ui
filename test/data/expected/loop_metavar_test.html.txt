Content-Type: text/plain
layer at (0,0) size 800x600
  RenderView at (0,0) size 800x600
layer at (0,0) size 800x124
  RenderBlock {HTML} at (0,0) size 800x124
    RenderBody {BODY} at (8,8) size 784x108
      RenderBlock (anonymous) at (0,0) size 784x18
        RenderText {#text} at (0,0) size 300x18
          text run at (0,0) width 300: "Test the $index meta variable:"
      RenderBlock {DIV} at (0,18) size 784x18
        RenderText {#text} at (0,0) size 10x18
          text run at (0,0) width 10: "\""
        RenderText {#text} at (10,0) size 30x18
          text run at (10,0) width 30: "foo"
        RenderText {#text} at (40,0) size 90x18
          text run at (40,0) width 90: "\" is the "
        RenderText {#text} at (130,0) size 10x18
          text run at (130,0) width 10: "0"
        RenderText {#text} at (140,0) size 20x18
          text run at (140,0) width 20: "th"
        RenderText {#text} at (160,0) size 130x18
          text run at (160,0) width 130: " item out of "
        RenderText {#text} at (290,0) size 10x18
          text run at (290,0) width 10: "5"
        RenderText {#text} at (300,0) size 10x18
          text run at (300,0) width 10: "."
      RenderBlock {DIV} at (0,36) size 784x18
        RenderText {#text} at (0,0) size 10x18
          text run at (0,0) width 10: "\""
        RenderText {#text} at (10,0) size 30x18
          text run at (10,0) width 30: "bar"
        RenderText {#text} at (40,0) size 90x18
          text run at (40,0) width 90: "\" is the "
        RenderText {#text} at (130,0) size 10x18
          text run at (130,0) width 10: "1"
        RenderText {#text} at (140,0) size 20x18
          text run at (140,0) width 20: "st"
        RenderText {#text} at (160,0) size 130x18
          text run at (160,0) width 130: " item out of "
        RenderText {#text} at (290,0) size 10x18
          text run at (290,0) width 10: "5"
        RenderText {#text} at (300,0) size 10x18
          text run at (300,0) width 10: "."
      RenderBlock {DIV} at (0,54) size 784x18
        RenderText {#text} at (0,0) size 10x18
          text run at (0,0) width 10: "\""
        RenderText {#text} at (10,0) size 30x18
          text run at (10,0) width 30: "baz"
        RenderText {#text} at (40,0) size 90x18
          text run at (40,0) width 90: "\" is the "
        RenderText {#text} at (130,0) size 10x18
          text run at (130,0) width 10: "2"
        RenderText {#text} at (140,0) size 20x18
          text run at (140,0) width 20: "nd"
        RenderText {#text} at (160,0) size 130x18
          text run at (160,0) width 130: " item out of "
        RenderText {#text} at (290,0) size 10x18
          text run at (290,0) width 10: "5"
        RenderText {#text} at (300,0) size 10x18
          text run at (300,0) width 10: "."
      RenderBlock {DIV} at (0,72) size 784x18
        RenderText {#text} at (0,0) size 10x18
          text run at (0,0) width 10: "\""
        RenderText {#text} at (10,0) size 30x18
          text run at (10,0) width 30: "qux"
        RenderText {#text} at (40,0) size 90x18
          text run at (40,0) width 90: "\" is the "
        RenderText {#text} at (130,0) size 10x18
          text run at (130,0) width 10: "3"
        RenderText {#text} at (140,0) size 20x18
          text run at (140,0) width 20: "rd"
        RenderText {#text} at (160,0) size 130x18
          text run at (160,0) width 130: " item out of "
        RenderText {#text} at (290,0) size 10x18
          text run at (290,0) width 10: "5"
        RenderText {#text} at (300,0) size 10x18
          text run at (300,0) width 10: "."
      RenderBlock {DIV} at (0,90) size 784x18
        RenderText {#text} at (0,0) size 10x18
          text run at (0,0) width 10: "\""
        RenderText {#text} at (10,0) size 40x18
          text run at (10,0) width 40: "quux"
        RenderText {#text} at (50,0) size 90x18
          text run at (50,0) width 90: "\" is the "
        RenderText {#text} at (140,0) size 10x18
          text run at (140,0) width 10: "4"
        RenderText {#text} at (150,0) size 20x18
          text run at (150,0) width 20: "th"
        RenderText {#text} at (170,0) size 130x18
          text run at (170,0) width 130: " item out of "
        RenderText {#text} at (300,0) size 10x18
          text run at (300,0) width 10: "5"
        RenderText {#text} at (310,0) size 10x18
          text run at (310,0) width 10: "."
      RenderBlock (anonymous) at (0,108) size 784x0
        RenderText {#text} at (0,0) size 0x0
        RenderText {#text} at (0,0) size 0x0
